lib/new_relic/telemetry_sdk/harvester.rb

```diff
@@ -0,0 +1,66 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-telemetry-sdk-ruby/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module TelemetrySdk
+    class Harvester
+
+      attr_reader :interval
+
+      def initialize interval = 5
+        @interval = interval
+        @harvestables = {}
+        @shutdown = false
+        @running = false
+        @lock = Mutex.new
+      end
+
+      def register name, buffer, client
+        @lock.synchronize do 
+          @harvestables[name] = {
+            buffer: buffer,
+            client: client
+          }
+        end
+      end
+
+      def lookup name 
+        @harvestables[name]
+      end
+
+      def is_running?
+        @running
+      end
+      
+      def start
+        @running = true
+        @harvest_thread = Thread.new do
+          while !@shutdown do
+            sleep @interval
+            harvest
+          end
+          @running = false
+        end
+      end
+
+      def stop
```

Comment:
This is not mandated by the spec, but I think it'd be a good idea to harvest and send any remaining data when shutting down.

Reply:
That makes sense, I added a final call to the harvest method at the end of the harvest thread, so that it gets called one last time after stop is called.